### ISSUE TITLE
Problem to compare mixing extensions

### ISSUE THREAD

**Issue by koodee** [PerfectSlayer/SourceChecksum#1]
The problem is in ChecksumTool.java, function compareResource

if leftFile = toto and right file = toto.c, during the compare mixing extensions, leftSubname = toto and rightSubname = toto, so compare will output 0, even though these are not the same files.

The question is, why do you care about extensions, instead of doing resource1.getName().compareTo(resource2.getName()) ?

**Comment by PerfectSlayer** [comment]
Did you talk about [this case](https://github.com/PerfectSlayer/SourceChecksum/blob/master/src/main/java/fr/hardcoding/software/sourcechecksum/ChecksumTool.java#L359)?

I agree they should not be the same (for toto and toto.c).
Nevertheless, I should make tests without extension to settle this kind of cases:

``` java
System.out.println("toto.c".compareTo("toto1")); // -3
System.out.println("toto.c".compareTo("toto#")); // 11
```

Because toto file could not be before and after longer filename starting with toto.

I'll make a patch soon :wink:

**Comment by PerfectSlayer** [comment]
Fixed in 89e1d63993ab7523554e440717dc61a9b1812bc1
